Alignment/APEEstimation/test/plottingTools/trendPlotter.py

```diff
@@ -0,0 +1,307 @@
+import ROOT
+ROOT.gROOT.SetBatch(True)
+from setTDRStyle import setTDRStyle
+from granularity import *
+import os
+
+try:
+    base = os.environ['CMSSW_BASE']
+except KeyError:
+    base = "../../../../.."
+
+# 2016 is missing here
+lumiFiles = {
+    2017: "{base}/src/Alignment/APEEstimation/data/lumiperrun2017.txt",
+    2018: "{base}/src/Alignment/APEEstimation/data/lumiperrun2018.txt",
+}
+pixelIOVs = {
+    2016: [],
+    2017: [297281, 298653, 299443, 300389, 301046, 302131, 303790, 303998, 304911],
+    2018: [316758, 317527, 317661, 317664, 318227, 320377],
+}
+
+
+lumis = {}
+intLumis = {}
+runs = {}
+years = []
+
+for year in lumiFiles.keys():
+    runs[year] = []
+    years.append(year)
+    intLumis[year] = 0
+    with open(lumiFiles[year].format(base=base), "r") as fi:
+        for line in fi:
+            run, lumi = line.split(" ")
+            run = int(run)
+            lumi = float(lumi)*0.001 # convert to fb^-1
+            lumis[run] = lumi
+            runs[year].append(run)
+            intLumis[year] += lumi
+years.sort()
+
+def runToLumi(whichRun, fromYear, inclusive=False):
+    lumiResult = 0.0
+    for year in years:
+        if year < fromYear:
+            continue
+        for run in runs[year]:
+            if run > whichRun:
+                break
+            if run < whichRun:
+                lumiResult += lumis[run]
+            if run == whichRun and inclusive:
+                lumiResult += lumis[run]
+    return lumiResult
+
+
+def whichYear(run):
+    thisYear = -1
+    years = runs.keys()
+    years.sort()
+    for year in years:
+        if min(runs[year]) <= run:
+            thisYear = year
+    return thisYear
+    print "Run %d not in range of any year"%(run)
```

Comment:
@mteroerd this will fail python3 compilation

Reply:
Thanks for the comment, fixed.